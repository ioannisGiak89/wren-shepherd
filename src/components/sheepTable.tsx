import React from 'react';
import { DataGrid } from '@material-ui/data-grid';
import { Sheep } from '../types';

const columns = [
    { field: 'name', headerName: 'Name', width: 150 },
    { field: 'gender', headerName: 'Gender', width: 150 },
    { field: 'isBranded', headerName: 'Branded', width: 150 },
];

const SheepTable = ({ sheep }: SheepTableProps) => {
    const getRows = () =>
        Object.keys(sheep).map((id) => ({
            id,
            ...sheep[id],
        }));

    return (
        <div style={{ height: 400, width: '100%' }}>
            <DataGrid
                // onSelectionModelChange={handleSelect}
                rows={getRows()}
                columns={columns}
                pageSize={5}
                checkboxSelection
            />
        </div>
    );
};

interface SheepTableProps {
    sheep: { [key: string]: Sheep };
    // handleSelect: (param: GridSelectionModelChangeParams) => void;
}

export { SheepTable };
