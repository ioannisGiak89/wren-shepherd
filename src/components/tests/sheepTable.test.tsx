import React from 'react';
import renderer from 'react-test-renderer';
import { SheepTable } from '../sheepTable';
import { testData } from './testData';

describe('sheepTable', () => {
    it('should display sheep data', () => {
        const sheepTable = renderer
            .create(<SheepTable sheep={testData} handleSelect={jest.fn()} />)
            .toJSON();
        expect(sheepTable).toMatchSnapshot();
    });
});
