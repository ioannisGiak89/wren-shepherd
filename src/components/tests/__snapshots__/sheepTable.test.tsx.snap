// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sheepTable should display sheep data 1`] = `
<div
  style={
    Object {
      "height": 400,
      "width": "100%",
    }
  }
/>
`;
